### DISCUSSION TITLE
[分享] 浏览器脚本自动从 Cookie 获取 BDUSS 和 STOKEN

### DISCUSSION THREAD

**Discussion by RayPS** [qjfoidnh/BaiduPCS-Go#331] [Show and tell]
因为感到登录过程比较繁琐，所以写了个浏览器脚本，自动获取并复制所需 Cookie，以实现直接粘贴即可登录。
必须配合 [Tempermonkey Beta](https://www.tampermonkey.net/) 来使用，只有 Beta 版支持获取 Cookie。

### 声明权限
```
// ==UserScript==
// @match        https://pan.baidu.com/*
...
// @grant        GM_cookie
// @grant        GM_setClipboard
...
// ==/UserScript==
```

### 脚本
```js
    GM_cookie.list({ path: "/" }, function (cookies) {
        const bduss = cookies.find(({name}) => name === 'BDUSS')
        const stoken = cookies.find(({name}) => name === 'STOKEN')
        const login = ["login", "--bduss", bduss.value, "--stoken", stoken.value].join(" ")
        GM_setClipboard(login, "text", function () {
            console.log("已复制当前账号的登录命令到剪贴板：")
            console.info(login)
        })
    })
```
极简实现，凑合用，已经过测试，没有做意外处理等其他东西，欢迎讨论

**Comment by RayPS** [comment]
另外，这个项目是 Go 语言开发的，其实可以直接读取浏览器的 Cookies 自动登录，但是这个比较敏感，看开发者愿不愿意加入这个功能了。
https://github.com/browserutils/kooky

**Reply by Yangli93** [reply]
复制cookie后怎么使用？可以直接用cookie吗？

**Reply by RayPS** [reply]
看文档

**Comment by DavidYanFan** [comment]
> 由于登录过程比较繁琐，所以写了个浏览器脚本，自动获取并复制所需Cookie，以实现直接粘贴即可登录。 必须配合[Tempermonkey Beta](https://www.tampermonkey.net/)来使用，只有Beta版支持获取Cookie。
> 
> ### 声明权限
> ```
> // ==UserScript==
> // @match        https://pan.baidu.com/*
> ...
> // @grant        GM_cookie
> // @grant        GM_setClipboard
> ...
> // ==/UserScript==
> ```
> 
> ### 脚本
> ```js
>     GM_cookie.list({ path: "/" }, function (cookies) {
>         const bduss = cookies.find(({name}) => name === 'BDUSS')
>         const stoken = cookies.find(({name}) => name === 'STOKEN')
>         const login = ["login", "--bduss", bduss.value, "--stoken", stoken.value].join(" ")
>         GM_setClipboard(login, "text", function () {
>             console.log("已复制当前账号的登录命令到剪贴板：")
>             console.info(login)
>         })
>     })
> ```
> 
> 极简实现，凑合用，已经过测试，没有做意外处理等其他东西，欢迎讨论

爱了，可以用

**Comment by txhputting** [comment]
HI 请问下  BDUSS 和 STOKEN 需要频繁更新吗？浏览器看这两个信息有期限，BDUSS 一年，STOKEN 只有一个月；
